src/syscall_mock/mod.rs

```diff
@@ -133,26 +136,45 @@ impl SyscallMock {
         Ok(path)
     }
 
-    pub fn handle_end(mut self, exitcode: i32) -> TestResult {
+    pub fn handle_end(mut self, exitcode: i32, captured_stderr: Option<Vec<u8>>) -> TestResult {
         if let Some(expected_step) = self.protocol.steps.pop_front() {
-            self.register_error(&expected_step.command.format(), "<script terminated>");
+            self.register_step_error(&expected_step.command.format(), "<script terminated>");
         }
         if exitcode != self.protocol.exitcode {
-            self.register_error(
+            self.register_step_error(
                 &format!("<exitcode {}>", self.protocol.exitcode),
                 &format!("<exitcode {}>", exitcode),
             );
         }
+        if let Some(expected_stderr) = &self.protocol.stderr {
+            match captured_stderr {
+                None => panic!("stderr expected, but not captured"),
```

Comment:
maybe this should register an error similar to the message below? i.e.

```
    expected output to stderr: {:?}
    received no output to stderr
```

Reply:
This only happens if `check-protocols` has a bug: when the expected `stderr` is given in the protocol, but the script is started without capturing it. That's also why it's a `panic` instead of an `Err`. How can we make this more obvious? Maybe `panic!("check-protocols bug: stderr expected, but not captured")`?

Reply:
:open_mouth: :bulb:  aaaaaaah, that makes sense! I like that error message you proposed, that helps clarify that it's an internal problem rather than something related to the behavior of the subject under test :)